src/domain/innovation-hub/innovation.hub.service.ts

```diff
@@ -39,10 +43,45 @@ export class InnovationHubService {
       );
     }
 
-    const hub: IInnovationHub = InnovationHub.create(input);
+    const subdomainAvailable =
+      await this.namingService.isInnovationHubSubdomainAvailable(
+        createData.subdomain
+      );
+    if (!subdomainAvailable)
+      throw new ValidationException(
+        `Unable to create Innovation Hub: the provided subdomain is already taken: ${createData.subdomain}`,
+        LogContext.INNOVATION_HUB
+      );
+
+    if (createData.nameID && createData.nameID.length > 0) {
+      const nameAvailable =
+        await this.namingService.isInnovationHubNameIdAvailable(
+          createData.nameID
+        );
+      if (!nameAvailable)
+        throw new ValidationException(
+          `Unable to create Innovation Hub: the provided nameID is already taken: ${createData.nameID}`,
+          LogContext.INNOVATION_HUB
+        );
+    } else {
+      for (let i = 0; i < 9; i++) {
+        createData.nameID = this.namingService.createNameID(
+          `${createData.profileData.displayName}`
+        );
+        const nameAvailable =
+          await this.namingService.isInnovationHubNameIdAvailable(
+            createData.nameID
+          );
+        if (nameAvailable) break;
```

Comment:
after 9 unsuccessful iterations the nameID is still going to be set to an invalid value